scale-testing/pfioh_scale/test_pfioh_request.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+
+import sys, os
+sys.path.append('/home/ckaubisch/ChRIS-E2E/scale-testing')
+from test_request import test_request
+from scale_threads import scale_threads
+import time
+import threading
+import matplotlib.pyplot as plt
+import numpy as np
+from subprocess import Popen, PIPE
+import test_setup
+import uuid
+
+
+"""
+Performs scalability and performance tests by making increasingly more concurrent push/pull requests to pfioh and measuring resultant resource utilization and success rate
+"""
+
+RANGE = 50                                 # Number of concurrent requests to scale up to
+CAPTURE = True                            # Dictates whether or not to capture CPU and memory utilization
+PID = 0                                    # Process ID of pfioh
+
+pushpull="push"                            # Whether to perform a push or pull operation
+size="small"                               # Size of the directory (note: hardcoded; small=1MB, medium=25MB, large=100GB, xlarge=1GB) to push/pull
+#IP="pfioh-chris-scale.openapp.cloud"       # # IP & port on which pfioh is listening
+IP="pfioh-myproject.127.0.0.1.nip.io"
+
+nums = []                                  # Number of concurrent requests being made to pfioh
+cpu = []                                   # CPU utilization
+mem = []                                   # Memory utilization
+thr = []                                   # Throughput
+dur = []                                   # Duration
+s_rate = []                                # Success rate
+
+
+# Create test files and directories to be pulled/pushed if they don't already exist                                                                                                      
+if not (os.path.exists('/tmp/small') and os.path.exists('/tmp/medium') and os.path.exists('/tmp/large')):
+    test_setup.create()
```

Comment:
Probably better to handle these checks inside test_setup.create and make it idempotent.  Which it would already need to be given the way it's called.